src/Symfony/Component/Serializer/SerializerInterface.php

```diff
@@ -20,6 +24,10 @@ interface SerializerInterface
      * Serializes data in the appropriate format.
      *
      * @param array<string, mixed> $context Options normalizers/encoders have access to
+     *
+     * @throws NotEncodableValueException    Occurs when the format is not supported for deserialization
```

Comment:
do we need to document this one? Isn't this a programmer's mistake?
```suggestion
     * @throws NotEncodableValueException    Occurs when the format is not supported for serialization
```

Reply:
I dunno. I saw a similar exception as documented in NormalizerInterface phpdoc
https://github.com/symfony/symfony/blob/18d08ffe9c3a5ea93358c15845f224fb020b98a9/src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php#L33

So I thought it could be ok to be documented.

Also, unlike EncoderInterface which provide supportsEncoding which allows to write
```
if ($encoder->supportsEncoding()) {
    $encoder->encode();
}
```

there is no `supportSerialization` so when working with the serializerInterface we cannot write
```
if ($serializerInterface->supportSerialization()) {
    $serializerInterface->encode();
}
```
and may have to deal with
```
try {
     $serializerInterface->serialize();
} catch (NotEncodableValueException) {
     // Not encodable
}
```

Reply:
I remove it (and also remove it from the NormalizerInterface, where it doesn't fit what I find useful there, per the definition above)

Reply:
What about changing this to
```
@throws UnexpectedValueException      Occurs when a value cannot be encoder
```

There multiple UnextedValueException which are possibly thrown by the ENcoderInterface cf
https://github.com/symfony/symfony/blob/18d08ffe9c3a5ea93358c15845f224fb020b98a9/src/Symfony/Component/Serializer/Encoder/EncoderInterface.php#L28

Reply:
works for me

Reply:
Done :)